### PR TITLE
Update demo site

### PR DESCRIPTION
**Pull request by sibiraj-s**
Update demo site. Page now uses alpine.js.

### CODE REVIEW

**Comment by sibiraj-s** [comment]
@DanielRuf . do you have any opinions on using standardjs, prettier migrating the build to rollup etc?

File: assets/main.js [hunk 1]
```diff
@@ -0,0 +1,311 @@
+import Alpine from 'https://cdn.jsdelivr.net/npm/alpinejs@3/dist/module.esm.js'
```
Comment (DanielRuf) [inline assets/main.js:1]: Personally I'm not a big fan of CDNs regarding availability, security and data (GDPR).

**Comment by DanielRuf** [comment]
alpinejs is something I'm currently working with at work.

Can you describe the motivation and advantages of these changes? Just so we know why we decided to do this step.

> @DanielRuf . do you have any opinions on using standardjs, prettier, migrating the build to rollup etc?

Are these things which make contributions easier? Personally I don't like prettier (well, it is opiniated) but maybe I am just too old and not a fan of new things.

Do these solutions as setup help to make faster and easier releases? I'm happy to discuss the best way forward.

**Comment by DanielRuf** [comment]
I know that Grunt, browserify and so on are a few old solutions. And I have used rollup in another project fresh from the start. But will the migration work? You may know better than I do.

File: assets/main.js (hunk 1 above)
Comment (sibiraj-s) [reply assets/main.js:1]: I thought build system was too much for this and went with cdn's.

**Comment by sibiraj-s** [comment]
> I know that Grunt, browserify and so on are a few old solutions. And I have used rollup in another project fresh from the start. But will the migration work? You may know better than I do.

There should be no issues during migration and also maintaining compatibility should be possible as well. and we can provide multiple builds as well, such as esm, cjs for modern systems.

> Do these solutions as setup help to make faster and easier releases? I'm happy to discuss the best way forward.

It's the usual, having items like standardjs config enables more rules, catch obvious errors and etc. I chose standard.js because it is widely used, popular, and not very strict by default and with eslint we can always extend. I will help catching lots of errors upfront even with contributions, definitely worth adding it.

And prettier(not a big fan either) is optional, thought it might help in maintaining codestyle with contributions.

> Can you describe the motivation and advantages of these changes? Just so we know why we decided to do this step.

A lot changed since this package originally written. I am thinking about rewriting the same in modern es6 syntaxes, modules etc. (obviously without compromising the compatibility). This will help us in long run to maintain the package and easier for contributors too. And this change is just a first step. Also not big fan of raw html templates, difficult to go through. since IE will officially be dead in few months. We can also leverage latest features like script modules etc.

File: assets/main.js (hunk 1 above)
Comment (DanielRuf) [reply assets/main.js:1]: Can we find a good and effective way to use alpine without CDN and make the build setup smaller?
I'm open for recommendations and ideas.

If the CDN solution is the best solution and others are ok with this, then we can maybe make this step.

In my personal and other projects I've tried to add these things, same for example in materialize: https://github.com/materializecss/materialize/tree/v1-dev/docs/js (the dependencies are shipped with the website).

**Comment by DanielRuf** [comment]
> and we can provide multiple builds as well, such as esm, cjs for modern systems.

Sounds good.

> It's the usual, having items like standardjs config enables more rules, catch obvious errors and etc. I chose standard.js because it is widely used, popular, and not very strict by default and with eslint we can always extend. I will help catching lots of errors upfront even with contributions, definitely worth adding it.
> 
> And prettier(not a big fan either) is optional, thought it might help in maintaining codestyle with contributions.

Ok, this makes definitely sense and is definetely a good way to make maintenance and contributions easier.

> And this change is just a first step.

Souns like a good plan =)

> since IE will officially be dead in few months. We can also leverage latest features like script modules etc

Definitely a good idea since this will lead to less issues as we will not support older Node and browser versions in the longterm and the overall ecosystem has to move on.

File: assets/main.js (hunk 1 above)
Comment (DanielRuf) [reply assets/main.js]: We can maybe add the alpinejs file (as fallback) if someone wants to selfhost the web version of the minifier (and can not use CDNs in the corporate network due to internal policies) or if something may happen with the used CDN domains (just an idea so in the worst case we can then switch to the included version).

Comment (sibiraj-s) [reply assets/main.js]: For now I used skypack, they are GDPR compliant.

May be I can download alpinejs to assets dir and just use. Later with rollup, may be I can try making alpine a devDependency.

Comment (sibiraj-s) [reply assets/main.js]: > In my personal and other projects I've tried to add these things, same for example in materialize: https://github.com/materializecss/materialize/tree/v1-dev/docs/js (the dependencies are shipped with the website).

can be done.

Comment (sibiraj-s) [reply assets/main.js]: Just a thought. I agree to all your concerns about cdn's as they are 100% valid. but personally I feel for Github pages and I wouldn't worry much about it.

Comment (sibiraj-s) [reply assets/main.js]: Shall I go ahead and merge this? may be if someone complains we can take the other fallback options available.

**Review by DanielRuf — APPROVED** [review]

File: assets/main.js (hunk 1 above)
Comment (DanielRuf) [reply assets/main.js]: You can go ahead. I'm fine with the current changes.

**Pull request merged**